charts/dns_canary/templates/cronjob.yaml

```diff
@@ -0,0 +1,93 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ .Values.name }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    app: {{ .Values.name }}
+spec:
+  schedule: {{ .Values.schedule | quote }}
+  concurrencyPolicy: Allow
+  failedJobsHistoryLimit: 1
+  successfulJobsHistoryLimit: 3
+  suspend: false
+  jobTemplate:
+    spec:
+      backoffLimit: 6
+      completions: 1
+      parallelism: 1
+      template:
+        metadata:
+          labels:
+            app: {{ .Values.name }}
+        spec:
+          {{- with .Values.imagePullSecrets }}
+          imagePullSecrets:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          serviceAccountName: {{ include "dns-canary.serviceAccountName" . }}
+          securityContext:
+            {{- toYaml .Values.podSecurityContext | nindent 12 }}
+          containers:
+            - name: {{ .Chart.Name }}
+              securityContext:
+                {{- toYaml .Values.securityContext | nindent 16 }}
+              image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+              imagePullPolicy: {{ .Values.image.pullPolicy }}
+              command:
+                - /bin/sh
+                - -c
+                - |
+                  set -e
+                  TARGET="http://{{ .Values.targetService }}"
+
+                  if curl --max-time 3 -sSf $TARGET; then
```

Comment:
I'm concerned that just curling the endpoint may not be enough to test DNS resolution. Might be better to use something like `nslookup`. Here's an article on debugging dns resolution in k8s: https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/ See if you can adapt that for this purpose.